pkg/odo/cli/registry/util/util.go

```diff
@@ -34,3 +35,11 @@ func IsSecure(registryName string) bool {
 
 	return isSecure
 }
+
+func IsGitBasedRegistry(url string) bool {
+	return strings.Contains(url, "github.com") || strings.Contains(url, "raw.githubusercontent.com")
```

Comment:
Do we support any other git registries? Gitlab for example?

Reply:
As @kadel said during planning, we DO NOT have any other official git registry, other than the hacky odo-devfiles/registry. In fact, the `support` for GitHub is basically the conversion of `github.com` to `raw.githubusercontent`. 

We could technically show a warning for GitLab too though that would mean having to do support tens to hundres of known git services.  It will really hard to determine if a registry is git based if we want to be more generic as we will have to rely on git checks. `.git` is generally hidden on most online git services, which means we will have to go for attempting to clone the repo (which either means having `git` on the user's system or using golang based git library). Either will be overengineering not to mention us having to download the repo during clone.

So I am only showing a warning for GitHub right now.